Applications/TextEditor/TextEditorWidget.cpp

```diff
@@ -32,6 +32,9 @@ TextEditorWidget::TextEditorWidget()
         m_document_dirty = true;
         if (!was_dirty)
             update_title();
+
+        m_editor->undo_action().set_enabled(m_editor->able_to_undo());
```

Comment:
This is the wrong place to do this, since it will only work in the `TextEditor` application, and not for other programs that use `GTextEditor`.
The right place to update the actions is inside `GTextEditor` itself, whenever we update the undo stack and/or index.